libraries/AP_UAVCAN/AP_UAVCAN.cpp

```diff
@@ -0,0 +1,497 @@
+
+                    const int node_start_res = node->start();
+                    if (node_start_res < 0) {
+                        printf("UAVCAN: node start problem\n\r");
+                    }
+
+                    gnss_fix = new uavcan::Subscriber<uavcan::equipment::gnss::Fix>(*node);
+                    const int gnss_fix_start_res = gnss_fix->start(gnss_fix_cb);
+                    if (gnss_fix_start_res < 0) {
+                        return false;
+                    }
+
+                    gnss_aux = new uavcan::Subscriber<uavcan::equipment::gnss::Auxiliary>(*node);
+                    const int gnss_aux_start_res = gnss_aux->start(gnss_aux_cb);
+                    if (gnss_aux_start_res < 0) {
+                        return false;
+                    }
+
+                    magnetic = new uavcan::Subscriber<uavcan::equipment::ahrs::MagneticFieldStrength>(*node);
+                    const int magnetic_start_res = magnetic->start(magnetic_cb);
+                    if (magnetic_start_res < 0) {
+                        return false;
+                    }
+
+                    air_data_sp = new uavcan::Subscriber<uavcan::equipment::air_data::StaticPressure>(*node);
+                    const int air_data_sp_start_res = air_data_sp->start(air_data_sp_cb);
+                    if (air_data_sp_start_res < 0) {
+                        return false;
+                    }
+
+                    air_data_st = new uavcan::Subscriber<uavcan::equipment::air_data::StaticTemperature>(*node);
+                    const int air_data_st_start_res = air_data_st->start(air_data_st_cb);
+                    if (air_data_st_start_res < 0) {
+                        return false;
+                    }
+
+                    act_out_array = new uavcan::Publisher<uavcan::equipment::actuator::ArrayCommand>(*node);
+                    act_out_array->setTxTimeout(uavcan::MonotonicDuration::fromMSec(20));
+                    act_out_array->setPriority(uavcan::TransferPriority::OneLowerThanHighest);
+
+                    esc_raw = new uavcan::Publisher<uavcan::equipment::esc::RawCommand>(*node);
+                    esc_raw->setTxTimeout(uavcan::MonotonicDuration::fromMSec(20));
+                    esc_raw->setPriority(uavcan::TransferPriority::OneLowerThanHighest);
+
+                    /*
+                     * Informing other nodes that we're ready to work.
+                     * Default mode is INITIALIZING.
+                     */
+                    node->setModeOperational();
+
+                    _initialized = true;
+
+                    printf("UAVCAN: init done\n\r");
+
+                    return true;
+                }
+            }
+        }
+
+        if (_initialized == true) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+void AP_UAVCAN::do_cyclic(void)
+{
+
+    uint32_t _servo_bm = 127; //SRV_Channels::get_can_servo_bm();
+    uint32_t _esc_bm = SRV_Channels::get_can_esc_bm();
```

Comment:
No. This must not be const.
Every send to Servo is taking several frames. The same goes with ESC.
I am giving the option to user to select what servo channels need to be published on network. This will greatly reduce load on bus and on CPU.